lib/vrf.c

```diff
@@ -581,13 +552,6 @@ int vrf_socket(int domain, int type, int protocol, vrf_id_t vrf_id,
 		flog_err_sys(EC_LIB_SOCKET, "%s: Can't switch to VRF %u (%s)",
 			     __func__, vrf_id, safe_strerror(errno));
 
-	if (ret > 0 && interfacename && vrf_default_accepts_vrf(type)) {
-		zlog_err("VRF socket not used since net.ipv4.%s_l3mdev_accept != 0",
-			  (type == SOCK_STREAM ? "tcp" : "udp"));
-		errno = EEXIST; /* not sure if this is the best error... */
-		return -2;
-	}
```

Comment:
I think the original code is correct - at least when running multiple vrfs in the l3mdev=1 case.  In this case only 1 socket is needed.  Testing shows this is the case.  (I don't recall where the 1st scoket is open, but testing shows that it is...)